### PR TITLE
Fixed #682 - AssemblySource throws ArgumentExceptions

### LINKED ISSUE #682: System.IO.FileLoadException when published as self-contained with Net Core 3.1
The following occurs with net core 3.1 and Caliburn.Micro 4.0.117-alpha.

When calling
`dotnet publish --configuration Release --runtime win10-x64 --self-contained=true`
the generated output will not be able to run and exits with:

` System.IO.FileLoadException: Could not load file or assembly 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Arena.Framework.Application.Bootstrapper`1.SelectAssemblies()
   at Caliburn.Micro.BootstrapperBase.StartRuntime()
   at Caliburn.Micro.BootstrapperBase.Initialize()
   at Arena.Framework.Application.Bootstrapper`1..ctor()
   at Arena.Applications.Client.Bootstrapper..ctor()
   --- End of inner exception stack trace ---
   at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)
   at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)
   at System.Windows.Application.LoadComponent(Object component, Uri resourceLocator)
   at Arena.Applications.Client.App.InitializeComponent()
   at Arena.Applications.Client.App.Main()`

The default build works (framework dependent), but this happens if I try to pack everything together. My Bootstrapper is located inside a separate Assembly and uses Autofac and loads all modules through a config json file.

### PR DESCRIPTION
**Pull request by beachwalker**
... for duplicate keys. This catches the ArgumentExeption and continues instead of breaking. Also includes the DesignTime part, even though this never occurred there but keeps code the same style in both methods.

### CODE REVIEW

**Comment by beachwalker** [comment]
Sorry, see my other issue #688 ... had to do by hand. How to run the checks locally? Works on my machine. ;-) How can I help to fix this?

**Comment by beachwalker** [comment]
Sorry, see my other issue #688 ... had to do by hand. How to run the checks locally? Works on my machine. ;-) How can I help to fix this? Don't know why net-core Bootstrapper has effects on the builds of Xamarin etc.

**Comment by beachwalker** [comment]
> Sorry, see my other issue #688 ... had to do by hand. How to run the checks locally? Works on my machine. ;-) How can I help to fix this? Don't know why net-core Bootstrapper has effects on the builds of Xamarin etc.

The build error was not my fault... so I reverted back to the original submit.

**Comment by beachwalker** [comment]
I would like to help you out but I'm not familar with the Xamarin platform. Do you need help there? Looking forward to see a new -alpha/-beta (-release?) version including my fix. ;-)

File: src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs [hunk 1]
```diff
@@ -83,7 +83,19 @@ public void Initialize()
         protected virtual void StartDesignTime()
         {
             AssemblySource.Instance.Clear();
-            AssemblySource.Instance.AddRange(SelectAssemblies());
+			
+			foreach (var assembly in SelectAssemblies())
```
Comment (nigel-sampson) [inline src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs:87]: Looks like the formatting is off here.

File: src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs [hunk 2]
```diff
@@ -83,7 +83,19 @@ public void Initialize()
...
+            {
+                try
+                {
+                    if (!AssemblySource.Instance.Contains(assembly))
+                        AssemblySource.Instance.Add(assembly);
+                }
+                catch (ArgumentException)
```
Comment (nigel-sampson) [inline src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs:94]: If we're checking the assembly source now what times would you expect it to throw this exception?

**Comment by nigel-sampson** [comment]
Actually wondering if we can push the check into `AssemblySource` itself and get the same protection across all platforms.

File: src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs (hunk 2 above)
Comment (beachwalker) [reply src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs:94]: Happens if throwing in all assemblies in the app folder without filtering when publishing self-contained.

File: src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs (hunk 1 above)
Comment (beachwalker) [reply src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs:87]: Looks good when looking at the file alone. I was not able to see in the file, only in the diff. I look for this tomorrow. As I wrote in the other issue I was not able to build/open using Visual Studio to check the solution. So pre-configured  settings from you may not applied here.

Comment (beachwalker) [reply src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs:87]: Looks good here https://github.com/beachwalker/Caliburn.Micro/blob/e07ea940ceb9d8b034333678357721ba48fc1315/src/Caliburn.Micro.Platform/Platforms/net46-netcore/Bootstrapper.cs#L87

**Comment by beachwalker** [comment]
Happy to see it there, but can not test the other platforms here. Maybe just go for TryAdd instead of Add?

**Comment by nigel-sampson** [comment]
Pushed a fix to `master`.

**Pull request closed without merging**